pkg/mon/metric_daemon.go

```diff
@@ -131,6 +134,25 @@ func (d *cwDaemon) AddDefaults(data ...*MetricDatum) {
 	}
 }
 
+func (d *cwDaemon) emptyChannel() {
```

Comment:
Can you try to close the channel instead of doing this sentinel workaround?

```go
// here
close(d.channel)

// in the Run method
   case closed, dat := <-d.channel:
      if closed {
         return
      }
      d.append(dat)					
```

Reply:
I think if we close the channel and there is still someone writing to it, we get a panic in that thread. As we don't want metrics to change the behavior of the program, I am not sure this is something we want to risk just because someone has a process still running (especially before we have a kernel with multiple stages)

Reply:
I agree, but.
The `channel` is a private field of our daemon. 
To prevent this we should add a check before we write in the channel.
(check if it is closed)

I think closing the channel is just a more elegant solution.
 It is the use case that feature was made for (channel closing).